crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -381,6 +457,234 @@ impl Provider for KvRedisProvider {
+
+            let current_config = config_response.get(1).ok_or_else(|| {
+                error!("Unexpected response format from Redis CONFIG GET");
+                anyhow::anyhow!("Unexpected response format from Redis CONFIG GET")
+            })?;
+
+            if !current_config.contains('K')
+                || !current_config.contains('E')
+                || !current_config.contains('A')
+            {
+                error!("Redis keyspace-notifications not properly configured");
+                return Err(anyhow::anyhow!(
+                    "Redis keyspace-notifications not properly configured! \
+                        Expected 'KEA' in settings, but got '{}'. \
+                        Please run: CONFIG SET notify-keyspace-events KEA",
+                    current_config
+                ));
+            }
+
+            let wrpc = Arc::new(wrpc);
+            let wrpc_for_task = wrpc.clone();
+
+            let mut watched_keys = self.watched_keys.write().await;
+
+            let config: HashMap<UniCase<&str>, &String> = config
+                .iter()
+                .map(|(k, v)| (UniCase::new(k.as_str()), v))
+                .collect();
+
+            // Look for watch configuration in the format "watch: SET@key:value,DEL@key"
+            if let Some(watch_config) = config.get(&UniCase::new("watch")) {
+                for watch_entry in watch_config.split(',') {
+                    let watch_entry = watch_entry.trim();
+                    if watch_entry.is_empty() {
+                        continue;
+                    }
+                    let parts: Vec<&str> = watch_entry.split('@').collect();
+                    if parts.len() != 2 {
+                        error!("Invalid watch entry format: {watch_entry}. Expected FORMAT@KEY[:VALUE]");
+                    }
+
+                    let operation = parts[0].trim().to_uppercase();
+                    let key_value = parts[1].trim();
+
+                    match operation.as_str() {
+                        "SET" | "set" => {
+                            let key_value_parts: Vec<&str> = key_value.split(':').collect();
+                            if key_value_parts.len() != 2 {
+                                error!("Invalid SET watch format: {key_value}. SET requires both KEY:VALUE");
+                            }
+                            watched_keys
+                                .entry(key_value_parts[0].to_string())
+                                .or_default()
+                                .insert(WatchedKeyInfo {
+                                    event_type: WatchEventType::Set,
+                                    value: Some(key_value_parts[1].to_string()),
+                                    target: target_id.to_string(),
+                                });
+                        }
+                        "DEL" | "del" => {
+                            if key_value.contains(':') {
+                                error!(
```

Comment:
So as to not spam every place this happens with a comment, can you make sure that, in most cases, data isn't interpolated into the log messages in all places in this file? This leads to high cardinality when receiving logs and so most data should be a field instead of interpolated